Tests/iaas/key-manager/check-for-key-manager.py

```diff
@@ -0,0 +1,81 @@
+"""Mandatory APIs checker
+This script retrieves the endpoint catalog from Keystone using the OpenStack
+SDK and checks whether a key manager APi endpoint is present.
+The script relies on an OpenStack SDK compatible clouds.yaml file for
+authentication with Keystone.
+"""
+
+import argparse
+import logging
+import os
+
+import openstack
+
+
+logger = logging.getLogger(__name__)
+
+
+def connect(cloud_name: str) -> openstack.connection.Connection:
+    """Create a connection to an OpenStack cloud
+    :param string cloud_name:
+        The name of the configuration to load from clouds.yaml.
+    :returns: openstack.connnection.Connection
+    """
+    return openstack.connect(
+        cloud=cloud_name,
+    )
+
+
+def check_presence_of_key_manager(cloud_name: str):
+    try:
+        connection = connect(cloud_name)
+        services = connection.service_catalog
+    except Exception as e:
+        print(str(e))
+        raise Exception(
+            f"Connection to cloud '{cloud_name}' was not successfully. "
+            f"The Catalog endpoint could not be accessed. "
+            f"Please check your cloud connection and authorization."
+        )
+
+    for svc in services:
+        svc_type = svc['type']
+        if svc_type == "key-manager":
+            # key-manager is present
+            return 0
+
+    # we did not find the key-manager service
+    logger.warning("There is no key-manager endpoint in the cloud.")
+    # we do not fail, until a key-manager MUST be present
+    return 0
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="SCS Mandatory IaaS Service Checker")
+    parser.add_argument(
+        "--os-cloud", type=str,
+        help="Name of the cloud from clouds.yaml, alternative "
+        "to the OS_CLOUD environment variable"
+    )
+    parser.add_argument(
+        "--debug", action="store_true",
+        help="Enable OpenStack SDK debug logging"
+    )
+    args = parser.parse_args()
+    openstack.enable_logging(debug=args.debug)
+
+    # parse cloud name for lookup in clouds.yaml
+    cloud = os.environ.get("OS_CLOUD", None)
+    if args.os_cloud:
+        cloud = args.os_cloud
+    assert cloud, (
+        "You need to have the OS_CLOUD environment variable set to your cloud "
+        "name or pass it via --os-cloud"
+    )
+
+    return check_presence_of_key_manager(cloud)
```

Comment:
Now that we include the requirement of giving `member`s access to the Key Manager API if present, we should move the corresponding test from the Roles Standard over to here: https://github.com/SovereignCloudStack/standards/blob/65d72d4a8d7cb8598133ddb4fd7563d75cc039f1/Tests/iam/iaas-roles/standard-roles-check.py#L136-L189

Reply:
> I started integrating the code until I stumbled about this:
> 
> ```
> keystoneauth1.exceptions.http.Unauthorized: Error authenticating with application credential: Application credentials cannot request a scope. (HTTP 401) (Request-ID: req-afca1cf7-0c4f-49d9-b716-44c7bb4f6f72)
> ```
> 
> Now we are investigating, whether there are other ways to ensure that a logged in user only has the member role within the specified project.

@josephineSei mhm, weird, I didn't test it myself, but this should work?

see: https://opendev.org/openstack/keystone/commit/52da4d0e129048d1b808bdde07364cde698cf475

Reply:
I have a devstack that already partly includes the new secure RBAC roles. They are enabled in Keystone (and Nova and Glance I think), but not in Barbican. Maybe that is the reason for this behavior. I can take another look into this.

Reply:
I now check the roles of the current user, when there is an admin or manager role present (or any custom role), the check for the creation of a secret will be skipped and a warning logged.